### PR TITLE
Migrating styleFrom API to new version.

### PR DESCRIPTION
**Pull request by Franreno**
Updates the styleFrom API to use `backgroundColor` and `foregroundColor` instead of deprecated methods (`primary`, `onPrimary`).

Fixes [#105724](https://github.com/flutter/flutter/issues/105724)

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.10.5+3
+
+* Migrating styleFrom API to new version.
```
Comment (stuartmorgan) [inline packages/camera/camera/CHANGELOG.md:3]: This needs to follow https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style, so it should be "Migrates".

Comment (stuartmorgan) [inline packages/camera/camera/CHANGELOG.md:3]: We also usually want to be specific in changelogs; "new version" doesn't provide enough information for the reader to know what exactly is being changed. We should also make it clear that only examples are being changed. How about:

```
* Migrates `styleFrom` usage in examples off of deprecated `primary` and `onPrimary` parameters.
```

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for doing this cleanup! Just a note of the CHANGELOG wording (which applies to all of them, so it may be easiest to `git revert 1fe1ba4a` and then re-run `update-release-info`.

File: packages/file_selector/file_selector_windows/example/lib/open_image_page.dart [hunk 2]
```diff
@@ -48,8 +48,8 @@ class OpenImagePage extends StatelessWidget {
           children: <Widget>[
             ElevatedButton(
               style: ElevatedButton.styleFrom(
-                backgroundColors: Colors.blue,
-                foregroundColors: Colors.white,
+                backgroundColor: Colors.blue,
+                foregroundColor: Colors.white,
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_windows/example/lib/open_image_page.dart:52]: Does this mean you did the migration by hand? If so, I would definitely encourage using `flutter fix` in the future when possible, as it's both easier, and avoids potential errors like this (or more subtle ones, like accidentally swapping colors during the migration).

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

File: packages/camera/camera/CHANGELOG.md (hunk 1 above)
Comment (Franreno) [reply packages/camera/camera/CHANGELOG.md]: Thanks for reminding me of this. I have fixed it on the latest commit.

Comment (Franreno) [reply packages/camera/camera/CHANGELOG.md]: I agree with you! I changed it to your suggestion. Thank you.

File: packages/file_selector/file_selector_windows/example/lib/open_image_page.dart (hunk 2 above)
Comment (Franreno) [reply packages/file_selector/file_selector_windows/example/lib/open_image_page.dart:52]: Yes I did it by hand. I didn't know about the `flutter fix` command lol.
Thanks for the suggestion!

**Comment by Franreno** [comment]
Thanks for all the suggestions, @stuartmorgan!
I have re-run the changelog command and pushed another commit with the changes you suggested. 
Let me know if I did something wrong and need changes!

**Review by tarrinneal — APPROVED** [review]

**Comment by tarrinneal** [comment]
All looks good to me, if you could manage the changelog conflicts, we can land this.

**Comment by tarrinneal** [comment]
override to land due to broken test for failures on main

**Pull request merged**